controller_manager/src/ros2_control_node.cpp

```diff
@@ -46,10 +47,14 @@ int main(int argc, char ** argv)
       RCLCPP_INFO(cm->get_logger(), "update rate is %d Hz", update_rate);
 
       while (rclcpp::ok()) {
+        std::chrono::system_clock::time_point begin = std::chrono::system_clock::now();
         cm->read();
         cm->update();
         cm->write();
-        std::this_thread::sleep_for(std::chrono::nanoseconds(1000000000 / update_rate));
+        std::chrono::system_clock::time_point end = std::chrono::system_clock::now();
+        std::this_thread::sleep_for(
+          std::chrono::nanoseconds(1000000000 / update_rate) -
```

Comment:
I've applied @destogl 's suggestions for brevity of discussion, @youtalk I think this could now produce some undefined behaviour by passing a negative number to `sleep_for`. Perhaps could use an `std::max(0, ...)`?

Reply:
Thank you for your suggestion. I've fixed it. https://github.com/ros-controls/ros2_control/pull/302/commits/ec8dc49d35135977165a1d46819469deaba2be61